src/schematools/types.py

```diff
@@ -406,16 +426,25 @@ def __init__(
         if not self["schema"].get("$schema", "").startswith("http://json-schema.org/"):
             raise ValueError("Invalid JSON-schema contents of table")
 
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> DatasetTableSchema:
+        """Create tableschema from a dict.
+
+        When this factory function is used, the _parent_schema argument is missing
+        and will be set to an empty DatasetSchema.
+        """
+        return cls(DatasetSchema({"id": "parent", "type": "dataset"}), obj)
+
     @property
-    def name(self) -> Optional[str]:
+    def name(self) -> str:
         return self.get("shortname", self.id)
 
     @property
     def has_shortname(self) -> bool:
         return self.get("shortname") is not None
 
     @property
-    def dataset(self) -> Optional[DatasetSchema]:
+    def dataset(self) -> DatasetSchema:
```

Comment:
moet dan ook weer anders

Reply:
Restored to Optional. And added extra checks for None in the code where `dataset` is used.